src/app/pages/swap/components/swap-asset-sheet/components/use-swap-asset-list.tsx

```diff
@@ -33,12 +34,38 @@ export function useSwapAssetList<T extends BaseSwapContext<T>>({
   const isBaseList = type === 'base';
   const isQuoteList = type === 'quote';
 
-  // Filter out selected asset from selectable assets
-  const selectableAssets = assets.filter(
-    asset =>
-      (isBaseList && asset.name !== values.swapAssetQuote?.name) ||
-      (isQuoteList && asset.name !== values.swapAssetBase?.name)
-  );
+  // Initialize valid pairs hook with enabled flag based on asset availability
+  const {
+    filterValidAssets,
+    isLoading: isValidPairsLoading,
+    isError: isValidPairsError,
+  } = useBitflowValidPairs(assets, {
+    enabled: assets.length > 0,
+  });
+
+  // Filter out selected asset and apply valid pairs filtering
+  const selectableAssets = (() => {
+    // First, filter out the currently selected asset
+    const assetsWithoutSelected = assets.filter(
+      asset =>
+        (isBaseList && asset.name !== values.swapAssetQuote?.name) ||
+        (isQuoteList && asset.name !== values.swapAssetBase?.name)
+    );
+
+    // Then apply valid pairs filtering
+    if (isBaseList && values.swapAssetQuote?.tokenId) {
+      // When selecting base asset, filter by assets that can be paired with the selected quote
+      return filterValidAssets(assetsWithoutSelected, values.swapAssetQuote.tokenId, 'base');
+    }
```

Comment:
Can we tone down the comments? I've always tried to promote the philosophy that, if you have to write this many comments, it's because your code isn't readable. 

Later, someone will come along and update the code, without updating the comments, and it gets very confusing. 

e.g.
```ts
function filterOutCurrentlySelectedAssets(asset: Asset) {
  return (isBaseList && asset.name !== values.swapAssetQuote?.name) ||
    (isQuoteList && asset.name !== values.swapAssetBase?.name);
}
const assetsWithoutSelected = assets.filter(filterOutCurrentlySelectedAssets)
```